crates/electrum/src/electrum_ext.rs

```diff
@@ -54,29 +55,19 @@ impl<K, A: Anchor> ElectrumUpdate<K, A> {
     ///
     /// Refer to [`ElectrumUpdate`].
     pub fn finalize(
-        self,
+        &mut self,
         client: &Client,
         seen_at: Option<u64>,
         missing: Vec<Txid>,
-    ) -> Result<WalletUpdate<K, A>, Error> {
+    ) -> Result<(), Error> {
         let new_txs = client.batch_transaction_get(&missing)?;
-        let mut graph_update = TxGraph::<A>::new(new_txs);
-        for (txid, anchors) in self.graph_update {
+        for tx in new_txs {
             if let Some(seen_at) = seen_at {
-                let _ = graph_update.insert_seen_at(txid, seen_at);
-            }
-            for anchor in anchors {
-                let _ = graph_update.insert_anchor(txid, anchor);
+                let _ = self.graph_update.insert_seen_at(tx.txid(), seen_at);
```

Comment:
I think this is wrong but not due to this PR. We should only set `seen_at` if the transaction is not anchored: @evanlinjin

Reply:
I think it's okay to have both. The transaction may get reorged out and the last_seen will become useful